src/local_scheduler/local_scheduler_algorithm.cc

```diff
@@ -112,16 +110,29 @@ struct SchedulingAlgorithmState {
   object_entry *remote_objects;
 };
 
+TaskQueueEntry TaskQueueEntry_init(TaskSpec *spec, int64_t task_spec_size) {
+  TaskQueueEntry elt;
+  elt.spec = (TaskSpec *) malloc(task_spec_size);
+  memcpy(elt.spec, spec, task_spec_size);
+  elt.task_spec_size = task_spec_size;
+  return elt;
+}
+
+void TaskQueueEntry_free(TaskQueueEntry *entry) {
+  TaskSpec_free(entry->spec);
```

Comment:
Since we're using `std::list<TaskQueueEntry>`, entries that are removed from the list will automatically get freed (in contrast to if we had a `std::list<TaskQueueEntry *>` like in a previous commit).